src/main/java/pl/coderslab/UserDAO/UserDAO.java

```diff
@@ -16,29 +16,32 @@
     private static final String FIND_ALL_USERS_QUERY = "SELECT id, name, email, password, passwordSalt FROM users";
     private static final String FIND_ALL_GROUP_USERS_QUERY = "SELECT u.id, u.name, email, password, passwordSalt FROM users u JOIN user_group ug ON u.id = ug.user_id WHERE ug.id = ?";
 
-    public User create(User user) {
+    public User create(User user) throws SQLException {
+        Connection conn = getConnection();
         try  {
-            Connection conn = getConnection();
+            int idx = 0;
             PreparedStatement statement = conn.prepareStatement(CREATE_USER_QUERY, Statement.RETURN_GENERATED_KEYS);
-            statement.setString(1, user.getName());
-            statement.setString(2, user.getEmail());
-            statement.setString(3, user.getPassword());
-            statement.setString(4, user.getPasswordSalt());
+            statement.setString(++idx, user.getName());
+            statement.setString(++idx, user.getEmail());
+            statement.setString(++idx, user.getPassword());
+            statement.setString(++idx, user.getPasswordSalt());
             statement.executeUpdate();
             ResultSet resultSet = statement.getGeneratedKeys();
             if (resultSet.next()) {
                 user.setId(resultSet.getInt(1));
             }
             return user;
         } catch (SQLException e) {
-            e.printStackTrace();
-            return null;
+            catchCode(e);
+            return user;
+        } finally {
+            closeConnection(conn);
```

Comment:
na zajęciach nie zamykaliśmy połączenia a sotoswaliśmy "try with resources" warto to użyć również i w tym przypadku ;)